pyface/ui/qt4/workbench/workbench_window_layout.py

```diff
@@ -342,12 +342,12 @@ def _qt4_editor_tab_spinner(self, editor, name, new):
                 editor.control, QtCore.Qt.red
             )
 
-    @on_trait_change("window:active_editor")
-    def _qt4_active_editor_changed(self, old, new):
+    @observe("window:active_editor")
+    def _qt4_active_editor_changed(self, event):
         """ Handle change of active editor """
         # Reset tab title to foreground color
-        if new is not None:
-            self._qt4_editor_area.setTabTextColor(new.control)
+        if event.new is not None:
+            self._qt4_editor_area.setTabTextColor(event.new.control)
```

Comment:
this is nit-picky and makes suggestions to the deprecated workbench code - up to you if you want to include this suggested change - 

```suggestion
        editor = event.new 
        if editor is not None:
            self._qt4_editor_area.setTabTextColor(editor.control)
```